### DISCUSSION TITLE
Great Plugin but would like few features

### DISCUSSION THREAD

**Discussion by digital-fortress** [mlebreuil/netbox-contract#114] [Ideas]
I've thoroughly tested this plugin for a full day and find it excellent. However, attempting to modify certain features proved challenging due to the complexity of the plugin exceeding my current knowledge.

My specific requests are as follows:

Tenant and Accounting dimensions should not be mandatory fields. Accounting dimensions serve no purpose in my usage. If left unset, could their values be hidden?
As we track circuits with annual payments, displaying monthly recurring costs is not preferred. Could the line simply display Cost / Billing Rate? or an option to select monthly or annually.

I require an option to select the Provider Account, with it being non-mandatory. The external reference can be adapted to accommodate selecting the provider account.

The contract associated with the provider should be visible in the provider field.

**Comment by mlebreuil** [comment]
_Tenant and Accounting dimensions should not be mandatory fields. Accounting dimensions serve no purpose in my usage. If left unset, could their values be hidden?_

This will be available in next version

**Comment by mlebreuil** [comment]
_I require an option to select the Provider Account, with it being non-mandatory. The external reference can be adapted to accommodate selecting the provider account._

You can use a custom field for this purpose.
The plugin can link contract to other types of objects than Circuit one.

**Comment by mlebreuil** [comment]
_As we track circuits with annual payments, displaying monthly recurring costs is not preferred. Could the line simply display Cost / Billing Rate? or an option to select monthly or annually._

The mrc value is used in other calculation. To make it simple for the time being, I have simply added a calculated yrc filed to the view.
Available in next release.

**Reply by digital-fortress** [reply]
It's uncommon for a company to maintain monthly recurring costs, isn't it? The issue with monthly payments arises when trying to manually divide the annual cost, as it often results in decimal points. For instance, my circuit charge amounts to $572 annually, and dividing it by 12 yields decimals. Therefore, I propose that the plugin permit annual payment inputs.

**Reply by mlebreuil** [reply]
Both options are now available

**Reply by digital-fortress** [reply]
last time I checked wasnt supported in the release. Will update and see it tomorrow
